Material ui additional properties

jeffothy opened the pull request:
### Reasons for making this change

Add support for addtional properties in material-ui theme #1724

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

ValYouW commented on packages/material-ui/src/FieldTemplate/FieldTemplate.tsx:
```diff
@@ -8,39 +8,67 @@ import List from "@material-ui/core/List";
+  schema,
 }: FieldTemplateProps) => {
+  if (hidden) {
+    return <div className="field-hidden">{children}</div>;
```
This is indeed a needed change but I think it's beyond the scope of this PR, better to handle this in a separate issue as this fix is not 100% (for example the "add button" is still visible, existing props still take visual space instead of `disaply: none` etc).

jeffothy replied:
OK, sounds good.

ValYouW commented on packages/material-ui/src/FieldTemplate/WrapIfAdditional.tsx:
```diff
@@ -0,0 +1,92 @@
+  };
+
+  if (!additional) {
+    return <div className={classNames}>{children}</div>;
```
I think it is better to return a Fragment here, it will prevent adding an unnecessary `<div>` with `form-group field` classes (which if I'm not mistaken relevant for bootstrap only)

jeffothy replied:
Will do.

ValYouW commented on packages/material-ui/src/FieldTemplate/WrapIfAdditional.tsx:
```diff
@@ -0,0 +1,92 @@
+    onKeyChange(target.value);
+
+  return (
+    <Grid container={true} key={`${id}-key`} alignItems="center">
```
Adding some spacing between the key/value fields will make it look nicer:
```suggestion
    <Grid container={true} key={`${id}-key`} alignItems="center" spacing={2}>
```

ValYouW commented on packages/material-ui/src/FieldTemplate/WrapIfAdditional.tsx:
```diff
@@ -0,0 +1,92 @@
+  required,
+  schema,
+}: WrapIfAdditionalProps) => {
+  const readonlyAsDisabled = true;
```
I think this can be removed... Indeed that `antd` theme uses this alot, but it is taken from the `formContext`, not sure how it is relevant here, or how `formContext` can be applied to Material theme

jeffothy replied:
I agree, will update.

ValYouW commented on packages/material-ui/src/ObjectFieldTemplate/ObjectFieldTemplate.tsx:
```diff
@@ -49,6 +57,17 @@ const ObjectFieldTemplate = ({
+              <AddButton
+                className="object-property-expand"
+                onClick={onAddClick(schema)}
+                disabled={false}
```
This should be disabled if prop is disabled `disabled={disabled || readonly}`

jeffothy replied:
:facepalm: Will fix.

epicfaace commented:
This appears to be a duplicate of #1982, so I think we should go ahead and close it and merge the other PR instead. @ValYouW does that sound good to you, or did you notice anything different in particular about this PR vs #1982?

jeffothy commented:
> This appears to be a duplicate of #1982, so I think we should go ahead and close this PR and merge #1982 instead, because I've already reviewed #1982. @ValYouW did you notice anything different in particular about this PR vs #1982? If not, I'll close this PR and we'll finish up the work on #1982.

@epicfaace I'm quite enjoying the review with @ValYouW but of course do not intend to ignore your preference here. Absolutely happy to yield to the project maintainer direction! :smiley:

jeffothy commented:
@ValYouW Thanks for the review comments, I appreciate it! I've updated the branch accordingly.
Also saw that you had added a test for object's with additionalProperties in your fork so I grabbed that and added a `Co-authored-by: Yuval <valyouw@gmail.com>` trailer to the commit message.

Please take a look at resolve the conversations above if you're happy with the updates.

epicfaace commented:
> @epicfaace I'm quite enjoying the review with @ValYouW but of course do not intend to ignore your preference here. Absolutely happy to yield to the project maintainer direction! 😃

Okay, it's also fine to continue working with this PR, as long as both PRs solve the problem!

ValYouW commented:
@epicfaace All looks good to me! @jeffothy I can't resolve the conversations... After this is merged you can create an issue and PR for the "hidden" thing...

jeffothy commented:
> @epicfaace All looks good to me! @jeffothy I can't resolve the conversations... After this is merged you can create an issue and PR for the "hidden" thing...

I have resolved the conversations. It looks like we just need the approving reviewer mark.

jeffothy commented:
@epicfaace Are you able to approve or have a missed anything here?
